core/src/main/java/org/frankframework/pipes/RegExPipe.java

```diff
@@ -0,0 +1,195 @@
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Tries to match the input against the provided regex.
+ *
+ * <pre>{@code
+ * 	<RegExPipe name="regExPipe" regex="^(.*?)(name!)$">
+ * 		<Forward name="then" path="EXIT" />
+ * 		<Forward name="else" path="EXIT" />
+ * 	</RegExPipe>
+ * }</pre>
+ *
+ * <p>Input:</p>
+ * <pre>{@code
+ * 	Hello name!
+ * }</pre>
+ *
+ * <p>Output:</p>
+ * <pre>{@code
+ *  <matches>
+ * 		<match index="1" value="Hello name!">
+ * 			<group index="1">Hello </group>
+ * 			<group index="2">name!</group>
+ * 		</match>
+ *  </matches>
+ * }</pre>
+ *
+ * @ff.tip https://regex101.com can be used to quickly create and debug regex expressions.
+ *
+ */
+@Forward(name = RegExPipe.THEN_FORWARD, description = "When a match is found.")
+@Forward(name = RegExPipe.ELSE_FORWARD, description = "When no match is found.")
+public class RegExPipe extends AbstractPipe {
+
+	protected final static String THEN_FORWARD = "then";
+	protected final static String ELSE_FORWARD = "else";
+
+	enum RegExFlag {
+		CANON_EQ(Pattern.CANON_EQ),
+		CASE_INSENSITIVE(Pattern.CASE_INSENSITIVE),
+		COMMENTS(Pattern.COMMENTS),
+		DOT_ALL(Pattern.DOTALL),
+		LITERAL(Pattern.LITERAL),
+		MULTILINE(Pattern.MULTILINE),
+		UNICODE_CASE(Pattern.UNICODE_CASE),
+		UNICODE_CHARACTER_CLASS(Pattern.UNICODE_CHARACTER_CLASS),
+		UNIX_LINES(Pattern.UNIX_LINES);
+
+		private final @Getter int flag;
+
+		RegExFlag(int flag) {
+			this.flag = flag;
+		}
+	}
+
+	private String regex = "";
+	private String flags = "";
+
+	private Pattern pattern;
+
+	private PipeForward thenForward;
+	private PipeForward elseForward;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		final int flags = this.flags.isEmpty() ? 0 : Arrays.stream(this.flags.split(","))
+				.map(flag -> RegExFlag.valueOf(flag).getFlag())
+				.reduce(0, (a, b) -> a | b);
+
+		if (this.regex == null || this.regex.isEmpty()) {
+			throw new ConfigurationException("Value for `regex` is mandatory.");
+		}
+
+		this.pattern = Pattern.compile(this.regex, flags);
+
+		this.thenForward = findForward(THEN_FORWARD);
+		this.elseForward = findForward(ELSE_FORWARD);
+
+		if (this.thenForward == null) {
+			throw new ConfigurationException("has no forward with name [" + THEN_FORWARD + "]");
+		}
+
+		if (this.elseForward == null) {
+			throw new ConfigurationException("has no forward with name [" + ELSE_FORWARD + "]");
+		}
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
+		String sInput;
+		if (Message.isEmpty(message)) {
+			sInput = "";
+		} else {
+			try {
+				sInput = message.asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, "cannot open stream", e);
+			}
+			if (sInput == null) {
+				throw new PipeRunException(this, "Input message is empty");
+			}
+		}
+
+		Matcher matcher = this.pattern.matcher(sInput);
```

Comment:
indien `sInput` een lege string is, wil je dan wel matchen?

Reply:
Met `(.*)` kan je een lege string matchen. Ik betwijfel of dit echt gebruikt gaat worden, maar het is wel iets wat mogelijk moet zijn, dus ik laat het zo.